hexrd/ui/main_window.py

```diff
@@ -901,6 +912,10 @@ def on_show_raw_zoom_dialog(self):
         dialog.zoom_height = int(img.shape[0] / 5)
 
     def change_image_mode(self, mode):
+        # The active canvas change needs to be triggered *before* the image
+        # mode is changed. This makes sure that in-progress masks are completed
+        # and associated with the correct image mode.
+        self.active_canvas_changed()
```

Comment:
This will call `update_drawn_mask_line_picker_canvas()` since the active canvas did not necessarily change.

This is just helping us to make sure that if we have a function called `active_canvas_changed()`, that it only gets called when the active canvas actually changes.

Reply:
Fixed